AirCasting/SessionViews/ShareSession/GenerateSessionFileController.swift

```diff
@@ -0,0 +1,71 @@
+// Created by Lunar on 12/01/2022.
+//
+
+import Foundation
+import SSZipArchive
+
+protocol GenerateSessionFileController {
+    func generateFile(for session: SessionEntity) -> Result<URL, Error>
+}
+
+struct DefaultGenerateSessionFileController: GenerateSessionFileController {
+    let fileGenerator: CSVFileGenerator
+    
+    init() {
+        fileGenerator = DefaultCSVFileGenerator()
+    }
+    
+    func generateFile(for session: SessionEntity) -> Result<URL, Error> {
```

Comment:
we want to stop using core data entities everywhere, we should use the `Session` struct (see `Session.swift`) instead in new code unless it's strictly a database code.